packages/flutter_tools/lib/src/test/coverage_collector.dart

```diff
@@ -49,21 +49,24 @@ class CoverageCollector extends TestWatcher {
     // Synchronization is enforced by the API contract. Error handling
     // synchronization is done in the code below where `exitCode` is checked.
     // Callback cannot throw.
-    process.exitCode.then<Null>((int code) { // ignore: unawaited_futures
+    final Future<void> processFuture = process.exitCode.then<Null>((int code) { // ignore: unawaited_futures
       exitCode = code;
     });
     if (exitCode != null)
       throw new Exception('Failed to collect coverage, process terminated before coverage could be collected.');
 
     printTrace('pid $pid: collecting coverage data from $observatoryUri...');
-    final Map<String, dynamic> data = await coverage
+    Map<String, dynamic> data;
+    final Future<void> coverageFuture = coverage
         .collect(observatoryUri, false, false)
+        .then((Map<String, dynamic> result) { data = result; })
         .timeout(
           const Duration(minutes: 2),
           onTimeout: () {
             throw new Exception('Timed out while collecting coverage.');
           },
         );
+    await Future.any<void>(<Future<void>>[coverageFuture, processFuture]);
```

Comment:
actually we don't really want to do that, because it will remove the logic that catches the process ending early, i think... or at least make it dead code

Reply:
The goal is for this future to complete immediately if the flutter_tester process exits without producing coverage results (e.g. due to a crash).

Without this change, if flutter_tester crashes then the coverage future will not complete until the timeout expires.

Reply:
Did you ever see processFuture terminate? i'm testing a change that should be equivalent to yours and it doesn't seem to ever complete processFuture...

Reply:
The original bug involved a segfault in flutter_tester.  processFuture would complete when the flutter_tester process crashed.

Reply:
totally my fault, i copied that as "Future.all" instead of "Future.any", d'oh.